app/src/Repository/MsgrcptSearchRepository.php

```diff
@@ -91,19 +103,56 @@ public function countByTypeAndDays(
 
         $query = $queryBuilder->getQuery();
 
+        $query->enableResultCache(3600);
+
         return $query->getScalarResult();
     }
 
     protected function getBaseQueryBuilder(): QueryBuilder
     {
+        // $queryBuilder = $this->createQueryBuilder('m');
+
         $queryBuilder = $this->createQueryBuilder('mr');
         $queryBuilder->select('mr')
-            ->leftJoin('App\Entity\Msgs', 'm', Join::WITH, 'm.mailId = mr.mailId AND m.partitionTag = mr.partitionTag')
-            ->leftJoin('App\Entity\Maddr', 'maddr', Join::WITH, 'maddr.id = mr.rid');
+            ->innerJoin('App\Entity\Msgs', 'm', Join::WITH, 'm.mailId = mr.mailId AND m.partitionTag = mr.partitionTag')
+            ->innerJoin('App\Entity\Maddr', 'maddr', Join::WITH, 'maddr.id = mr.rid');
+
 
         return $queryBuilder;
     }
 
+    private function createCountQueryBuilder(
+        QueryBuilder $baseQueryBuilder,
+        ?User $user,
+        ?string $searchKey,
+        ?int $fromDate
+    ): QueryBuilder {
+        $countQueryBuilder = clone $baseQueryBuilder;
+
+        $hasFilters = !empty($searchKey) || $fromDate !== null;
+
+        if ($hasFilters) {
+            $countQueryBuilder->resetDQLPart('orderBy');
+            $countQueryBuilder->select('COUNT(mr.mailId)');
+
+            return $countQueryBuilder;
+        }
+
+        $countQueryBuilder->resetDQLPart('join');
+        $countQueryBuilder->resetDQLPart('from');
+        $countQueryBuilder->resetDQLPart('orderBy');
+
+        $countQueryBuilder
+            ->from(MsgRcpt::class, 'mr')
```

Comment:
Tiens, c’est pas sensible à la casse ? 🤔 

```suggestion
            ->from(Msgrcpt::class, 'mr')
```